receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -39,8 +42,9 @@ type prometheusRemoteWriteReceiver struct {
 	settings     receiver.Settings
 	nextConsumer consumer.Metrics
 
-	config *Config
-	server *http.Server
+	jobInstanceCache map[string]pmetric.ResourceMetrics
```

Comment:
Do we have any mechanism to expire items in this cache?  We don't want this to leak memory.

Actually, i'm not sure we should be storing the job + instance cache on the receiver itself. We probably want this cache around for a single PRW request, right?  Otherwise, we will keep appending new metrics to the same resourceMetrics forever.

Reply:
I think we need to keep the cache around for more than just one PRW request, unfortunately :/

Since we can't guarantee that the `target_info` metric comes in all requests, we need to cache it and re-use it to build the resource attributes in subsequent requests

Reply:
I was planning to tackle `target_info` in another PR though, and Job+Instance would be just an intermediary step until I get there

Reply:
And agreed that we need to periodically expire the cache, I was hoping to deal with it in a follow up PR :)

Reply:
Actually, this won't work as-is.  You should cache the `Resource` across requests, but you should *not* be caching the `ResourceMetrics` across requests, as that contains all of the metrics as well.